src/Google/GlobalSiteTag.php

```diff
@@ -109,6 +138,73 @@ function () {
 				$this->display_page_view_event_snippet();
 			}
 		);
+
+		$this->product_data_hooks();
+		$this->register_assets();
+	}
+
+	/**
+	 * Attach filters to add product data required for tracking events.
+	 */
+	protected function product_data_hooks() {
+		// Add product data for any add_to_cart link.
+		add_filter(
+			'woocommerce_loop_add_to_cart_link',
+			function ( $link, $product, $args ) {
+				$this->add_product_data( $product );
+				return $link;
+			},
+			10,
+			3
+		);
```

Comment:
💅 / ❓  

Since we are not really filtering or modifying the data. I wonder if some hook would be more appropiate.

Reply:
Do you have a suggestion of which action hook to use instead? Both actions and filters are hooks.

Reply:
> Both actions and filters are hooks.

Yes, sorry I was not precise, I meant `action` as far as I know `filter` is to modify the value and return that value whereas `action` allows to run some task when something happens (I think this is the case) 

> Do you have a suggestion of which action hook to use instead?

Unfortunately I'm not familiar with the available hooks in this part of the codebase. But I can investigate further 👌

Reply:
Yes, that's correct although they are still very similar and it's quite common in other code to hook into a filter if there is no action hook available, even if we don't change the output.

The most convenient way is to use this filter, the only other way I found was to possibly hook into when it's loading the template. But with themes and plugins overriding templates, I felt it was much simpler to just stick with the filter.